src/core/manager.c

```diff
@@ -1111,11 +1125,11 @@ static unsigned manager_dispatch_gc_unit_queue(Manager *m) {
         while ((u = m->gc_unit_queue)) {
                 assert(u->in_gc_queue);
 
-                unit_gc_sweep(u, gc_marker);
-
                 LIST_REMOVE(gc_queue, m->gc_unit_queue, u);
                 u->in_gc_queue = false;
 
+                unit_gc_sweep(u, gc_marker);
+
```

Comment:
hmm, why move this down? is there a strong reason? (just wondering, not saying it shouldn't be moved, but i was wondering if there's a logical reason behind this move)

Reply:
Yeah, I should have mentioned this in the commit message. It seemed cleaner to me this way, but looking at this again, I don't see an advantage. I removed this chunk.